librz/debug/p/native/bsd/bsd_debug.c

```diff
@@ -541,6 +541,69 @@ RzList *bsd_desc_list(int pid) {
 
 	free(buf);
 	return ret;
+#elif __NetBSD__
+	RzList *ret = NULL;
+	char path[512], file[512], buf[512];
+	struct dirent *de;
+	RzDebugDesc *desc;
+	int type, perm;
+	int len, len2;
+	struct stat st;
+	DIR *dd = NULL;
+
+	snprintf(path, sizeof(path), "/proc/%i/fd/", pid);
+	if (!(dd = opendir(path))) {
+		rz_sys_perror("opendir /proc/x/fd");
+		return NULL;
+	}
+	ret = rz_list_new();
+	if (!ret) {
+		closedir(dd);
+		return NULL;
+	}
+	ret->free = (RzListFree)rz_debug_desc_free;
+	while ((de = (struct dirent *)readdir(dd))) {
+		if (de->d_name[0] == '.') {
+			continue;
+		}
+		len = strlen(path);
+		len2 = strlen(de->d_name);
+		if (len + len2 + 1 >= sizeof(file)) {
+			rz_list_free(ret);
+			closedir(dd);
+			eprintf("Filename is too long");
+			return NULL;
```

Comment:
maybe it is better to just have a `goto fail;` and have this (excluding the error message that can stay here) at the end.
also for errors use `RZ_LOG_ERROR`, not eprintf.

Reply:
yes I just copied the linux version :) will update...

Reply:
this needs to be 100% refactored then on linux too